File: 2540-minimum-common-value/2540-minimum-common-value.go
func getCommon(nums1 []int, nums2 []int) int {
	index1 := 0
	index2 := 0
	for {
		if index1 >= len(nums1) || index2 >= len(nums2) {
			return -1
		}

		num1 := nums1[index1]
		num2 := nums2[index2]
		
		if num1 == num2 {
			return num1
		}
		
		if num1 > num2 {
			index2++
		} else {
			index1++
		}
	}
}

File: 2540-minimum-common-value/2540-minimum-common-value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        int index1 = 0;
        int index2 = 0;

        while (true) {
            if (index1 >= nums1.length || index2 >= nums2.length) {
                return -1;
            }
            
            int num1 = nums1[index1];
            int num2 = nums2[index2];
            
            if (num1 == num2) {
                return num1;
            } else if (num1 < num2) {
                index1++;
            } else {
                index2++;
            }
        }
    }
}